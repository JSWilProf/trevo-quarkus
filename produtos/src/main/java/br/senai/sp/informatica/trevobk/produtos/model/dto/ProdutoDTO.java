package br.senai.sp.informatica.trevobk.produtos.model.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Set;

@Data
@EqualsAndHashCode(callSuper = false)
public class ProdutoDTO {
    private String nome;
    private String descricao;
    private String area;
    private Set<String> culturas;
}
