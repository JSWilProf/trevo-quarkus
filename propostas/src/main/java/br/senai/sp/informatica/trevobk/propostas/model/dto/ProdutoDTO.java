package br.senai.sp.informatica.trevobk.propostas.model.dto;

import lombok.Data;

import java.util.Set;

@Data
public class ProdutoDTO {
    private String nome;
    private String descricao;
    private String area;
    private String imagem;
    private Set<String> culturas;
    private String cadastro;
    private String status;
}
