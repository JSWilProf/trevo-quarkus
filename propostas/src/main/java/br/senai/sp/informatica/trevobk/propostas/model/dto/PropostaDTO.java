package br.senai.sp.informatica.trevobk.propostas.model.dto;

import javax.validation.constraints.*;
import lombok.Data;

@Data
public class PropostaDTO {
    @NotEmpty(message = "O nome é obrigatório")
    private String nome;
    @Email(message = "O e-mail é inválido")
    @NotEmpty(message = "O e-mail é obrigatório")
    private String email;
    @Pattern(regexp = "(\\([1-9][0-9]\\) )?9?[0-9]{4}-[0-9]{4}", message = "O nº do telefone é inválido")
    @NotEmpty(message = "O telefone é obrigatório")
    private String telefone;
    @Min(value = 1, message = "O código do produto é obrigatório")
    @NotNull(message = "O código do produto é obrigatório")
    private Long idProduto;
}
