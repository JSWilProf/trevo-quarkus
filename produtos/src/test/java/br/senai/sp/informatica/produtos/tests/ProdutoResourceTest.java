package br.senai.sp.informatica.produtos.tests;

import br.senai.sp.informatica.produtos.utils.TokenUtils;
import br.senai.sp.informatica.trevobk.produtos.model.dto.ProdutoDTO;
import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.http.Header;
import io.restassured.specification.RequestSpecification;
import org.approvaltests.JsonApprovals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.Set;

// IMPORTANTE: Para execução dos testes é necessário que o flyway esteja ativado

@QuarkusTest
@QuarkusTestResource(ProdutoTestLifecycleManager.class)
public class ProdutoResourceTest {
    private String token;

    @BeforeEach
    public void geraToken() throws Exception {
        token = TokenUtils.generateTokenString("/JWTProprietarioClaims.json", null);
    }

    private RequestSpecification given() {
        return RestAssured.given()
            .contentType(ContentType.JSON)
            .header(new Header("Authorization", "Bearer " + token));
    }

    @Test
    public void testBuscarProdutos() {
        String resultado = given()
            .when().get("/api/produto/{idProduto}", 1)
            .then()
            .statusCode(200)
            .extract().asString();
        JsonApprovals.verifyJson(resultado);
    }

    @Test
    public void testSalvarProduto() {
        var produto = new ProdutoDTO();
        produto.setNome("Galaga II XP");
        produto.setDescricao("Melhor Performance nas Aplicações.");
        produto.setArea("Até 50 ha");
        produto.setCulturas(Set.of("Café", "Frutas"));

        String resultado = given()
            .header("Content-type", "application/json")
            .and()
            .body(produto)
            .when().post("/api/produto/")
            .then()
            .statusCode(200)
            .extract().asString();
        JsonApprovals.verifyJson(resultado);
    }
}
