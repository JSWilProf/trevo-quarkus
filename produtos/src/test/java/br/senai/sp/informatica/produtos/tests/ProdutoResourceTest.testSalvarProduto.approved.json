{
  "area": "Até 50 ha",
  "culturas": [
    "Frutas",
    "Café"
  ],
  "descricao": "Melhor Performance nas Aplicações.",
  "nome": "Galaga II XP"
}